Implementation of both UI and back-end for the customer login flow

danielalyoshin opened the pull request:
Implementation of both UI and back-end for the customer login flow.

Includes:
- Completed UI
- Account creation
- Initialization of user data in Firebase
- User authentication

danielalyoshin committed:
Implement UI for customer login flow

Co-authored-by: Murphy Lee <murphylee2004@gmail.com>

murphylee10 committed:
Implement back-end for customer login flow

- Implement account creation
- Implement initialization of user data in Firebase
- Implement user authentication

wobertt commented on app/src/main/res/layout/activity_customer_login.xml:
```diff
@@ -0,0 +1,132 @@
+
+    <Button
+        android:id="@+id/customer_login_activity_create_account_button"
+        android:layout_width="150dp"
```
"Create Account" text clips on default virtual device: https://i.snipboard.io/ze8ZtQ.jpg. Increasing width to 200dp seems to fix it.

wobertt commented on app/src/main/java/com/example/b07_project_team1/CustomerCreateAccount.java:
```diff
@@ -0,0 +1,124 @@
+}
+
+/*
+        ~DATABASE STRUCTURE~
```
I think we should leave the database structure comments out

wobertt commented on app/src/main/res/layout/activity_customer_create_account.xml:
```diff
@@ -0,0 +1,119 @@
+    android:background="@color/pure_white"
+    tools:context=".CustomerCreateAccount">
+
+    <TextView
```
Long message clips on default virtual device: https://i.snipboard.io/w75X0O.jpg. Not sure how to fix - I can't find the error message in the codebase.

wobertt commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,17 @@
+    <string name="repeat_password_entry_box_text">Repeat Password:</string>
+    <string name="create_account_button_text">Create Account</string>
+    <string name="create_account_empty_fields_text">You have empty fields.</string>
+    <string name="create_account_password_repeat_no_match_text">Passwords do not Match.</string>
```
Match -> match

wobertt reviewed (changes requested):
Found a few UI issues, but I'm loving the design overall

murphylee10 committed:
Add visual feedback to UI and fix some issues

- Add visual feedback to some UI elements
- Make some minor adjustments to UI
- Remove unnecessary comments

Co-authored-by: Daniel Alyoshin <daniel.alyoshin@gmail.com>
Co-authored-by: Robert Chung <robertchung1462@gmail.com>

ThunderRoar commented:
I would recommend having the error disappear after a few seconds or having a highlight the error if they make a mistake again. For example, when I tried to input a short password and then try a bit larger one, it would not notify the user that the password still needs to be changed. So thats something we should try changing how the error format would notify.

But overall the app works great.

jang-35 commented:
Everything looks good to me, the app is functional and looks great.
